test/tomo/operator/xray_trafo_test.py

```diff
@@ -30,28 +30,261 @@
 # Internal
 import odl
 from odl.util.testutils import almost_equal
+from odl.tomo import ASTRA_CUDA_AVAILABLE
+from odl.tomo.util.testutils import skip_if_no_astra_cuda
 
 
-@pytest.mark.xfail  # Expected to fail since scaling of adjoint is wrong.
-@pytest.mark.skipif("not odl.tomo.ASTRA_CUDA_AVAILABLE")
-def test_xray_trafo():
-    """Test discrete X-ray transformt using ASTRA with CUDA."""
+# TODO: increase test coverage and improve tests
+
+# DiscreteLp volume / reconstruction space
+xx = 5
+nn = 5
+# xx = 5.5
+# nn = 11
+# xx = 5.25
+# nn = 21
+discr_vol_space_2d = odl.uniform_discr([-xx] * 2, [xx] * 2, [nn] * 2,
+                                       dtype='float32')
+discr_vol_space_3d = odl.uniform_discr([-xx] * 3, [xx] * 3, [nn] * 3,
+                                       dtype='float32')
+
+# Angle grid
+angle_intvl = odl.Interval(0, 2 * np.pi) - np.pi / 4
+agrid = odl.uniform_sampling(angle_intvl, 4)
+astride = float(agrid.stride)
+num_angle = agrid.ntotal
+
+# Detector grid
+# yy = 11
+# mm = 11
+yy = 10.5
+mm = 2 * 21
+dparams_2d = odl.Interval(-yy, yy)
+dgrid_2d = odl.uniform_sampling(dparams_2d, mm)
+dparams_3d = odl.Rectangle([-yy, -yy], [yy, yy])
+dgrid_3d = odl.uniform_sampling(dparams_3d, [mm] * 2)
+
+# Distances
+src_radius = 1000
+det_radius = 500
+mag = (src_radius + det_radius) / src_radius
+
+# Slice index to print
+z_vol = np.round(discr_vol_space_3d.shape[2] / 2)
+y_proj = np.floor(dgrid_3d.shape[1] / 2)
+
+# Precision for adjoint test
+precision = 4
+
+# Print section
+print('\n')
+print('angle grid = {}'.format(agrid.points().transpose() / np.pi))
+print('magnification = ', mag)
+print('2D:\n shape = ', discr_vol_space_2d.shape)
+print(' size = ', discr_vol_space_2d.grid.size())
+print(' dgrid = ', dgrid_2d.points().transpose()[0])
+print(' vol_grid = ', discr_vol_space_2d.points()[::nn][:, 0])
+print('3D:\n shape = ', discr_vol_space_3d.shape)
+print(' size = ', discr_vol_space_3d.grid.size())
+print(' dgrid = ', dgrid_3d.points()[:2 * nn][:, 1])
+print(' vol_grid = ', discr_vol_space_3d.points()[:, 2][:nn])
+print('\n vol z index = ', z_vol, '\n proj y index = ', y_proj)
+
+
+@skip_if_no_astra_cuda
+def test_xray_trafo_cpu_parallel2d():
+    """2D parallel-beam discrete X-ray transform with ASTRA and CPU."""
+
+    dparams = dparams_2d
+    dgrid = dgrid_2d
+    discr_vol_space = discr_vol_space_2d
+
+    # Geometry
+    geom = odl.tomo.Parallel2dGeometry(angle_intvl, dparams, agrid, dgrid)
+
+    # X-ray transform
+    A = odl.tomo.XrayTransform(discr_vol_space, geom,
```

Comment:
This convention is my fault, we should go for `op` if it is general or `projector` in this case, imo.